scene/gui/grid_container.cpp

```diff
@@ -217,15 +218,24 @@ void GridContainer::_notification(int p_what) {
 					s.y++;
 				}
 
+				Point2 p;
 				if (rtl) {
-					Point2 p(col_ofs - s.width, row_ofs);
+					p = Point2(col_ofs - s.width, row_ofs);
 					fit_child_in_rect(c, Rect2(p, s));
 					col_ofs -= s.width + theme_cache.h_separation;
 				} else {
-					Point2 p(col_ofs, row_ofs);
+					p = Point2(col_ofs, row_ofs);
 					fit_child_in_rect(c, Rect2(p, s));
 					col_ofs += s.width + theme_cache.h_separation;
 				}
+
+				if (row == 0 && col > 0) {
+					cell_sizes.append(p.x);
+				}
+
+				if (row > 0 && col == 0) {
```

Comment:
```suggestion
				} else if (row > 0 && col == 0) {
```
I'd say to help branching